kitsune/wiki/utils.py

```diff
@@ -194,3 +212,51 @@ def get_visible_document_or_404(
 
 def get_visible_revision_or_404(user, **kwargs):
     return get_object_or_404(Revision.objects.visible(user, **kwargs))
+
+
+def build_topics_data(request, product, topics):
+    """Build topics_data for use in topic cards"""
+    topics_data = []
+
+    # Get all documents for all topics up front
+    all_docs, _ = documents_for(
+        request.user, request.LANGUAGE_CODE, topics=list(topics), products=[product]
+    )
+
+    # Convert all docs to Document objects
+    doc_ids = [doc["id"] for doc in all_docs]
+    documents = (
+        Document.objects.filter(id__in=doc_ids)
+        .prefetch_related("topics")
+        .annotate(
+            topic_ids=StringAgg(
+                Cast("topics__id", TextField()), delimiter=",", ordering="topics__id", default=""
```

Comment:
from what I can tell, the ordering is not needed here. The way you are comparing the strings in the `topic_docs` below shouldn't be affected by removing the `ordering`.